pkg/router/router.go

```diff
@@ -26,50 +27,150 @@ import (
+// ExtensionLabelKey is the label key required by services to be registered as a
+// dashboard extension
 const ExtensionLabelKey = "tekton-dashboard-extension"
 
-// Label value required by services to be registered as a dashboard extension
+// ExtensionLabelValue is the label value required by services to be registered
+// as a dashboard extension
 const ExtensionLabelValue = "true"
 
-// Full label required by services to be registered as a dashboard extension
+// ExtensionLabel is the full label required by services to be registered as a
+// dashboard extension
 const ExtensionLabel = ExtensionLabelKey + "=" + ExtensionLabelValue
 
-// Annotation that specifies the valid extension path, defaults to "/"
-const ExtensionUrlKey = "tekton-dashboard-endpoints"
+// ExtensionURLKey specifies the valid extension paths, defaults to "/"
+const ExtensionURLKey = "tekton-dashboard-endpoints"
 
-// Delimiter to be used between the extension endpoints annotation value
+// ExtensionEndpointDelimiter is the Delimiter to be used between the extension
+// endpoints annotation value
 const ExtensionEndpointDelimiter = "."
 
-// Extension UI bundle location annotation
+// ExtensionBundleLocationKey IS the UI bundle location annotation key
 const ExtensionBundleLocationKey = "tekton-dashboard-bundle-location"
 
-// Extension display name annotation
+// ExtensionDisplayNameKey is the display name annotation key
 const ExtensionDisplayNameKey = "tekton-dashboard-display-name"
 
-// extensionRoot
+// ExtensionRoot is the URL root when accessing extensions
 const ExtensionRoot = "/v1/extensions"
 
 var webResourcesDir = os.Getenv("WEB_RESOURCES_DIR")
 
-func Register(resource endpoints.Resource) *restful.Container {
-	wsContainer := restful.NewContainer()
-	wsContainer.Router(restful.CurlyRouter{})
-
+// Register returns an HTTP handler that has the Dashboard REST API registered
+func Register(resource endpoints.Resource) *Handler {
 	logging.Log.Info("Registering all endpoints")
-	registerWeb(wsContainer)
-	registerEndpoints(resource, wsContainer)
-	registerPropertiesEndpoint(resource, wsContainer)
-	registerWebsocket(resource, wsContainer)
-	registerHealthProbe(resource, wsContainer)
-	registerReadinessProbe(resource, wsContainer)
-	registerExtensions(wsContainer)
-	registerKubeAPIProxy(resource, wsContainer)
-	return wsContainer
+	h := &Handler{
+		Container:       restful.NewContainer(),
+		uidExtensionMap: make(map[string]*Extension),
+	}
+
+	registerWeb(h.Container)
+	registerEndpoints(resource, h.Container)
+	registerPropertiesEndpoint(resource, h.Container)
+	registerWebsocket(resource, h.Container)
+	registerHealthProbe(resource, h.Container)
+	registerReadinessProbe(resource, h.Container)
+	registerKubeAPIProxy(resource, h.Container)
+	h.registerExtensions()
+	return h
+}
+
+// Handler is an HTTP handler with internal configuration to avoid global state
+type Handler struct {
+	*restful.Container
```

Comment:
Same question as above - do we definitely need all the explicit pointer refs?

Reply:
I don't think we need to, but since the return type of `restful.NewContainer()` is `*restful.Container` and the register functions expect this type I embedded the pointer.